sound/soc/sof/ipc4-pcm.c

```diff
@@ -571,12 +571,12 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	}
 
 	/* Chain DMA does not use copiers, so no fixup needed */
-	if (!use_chain_dma) {
-		int ret = sof_ipc4_pcm_dai_link_fixup_rate(sdev, params, ipc4_copier);
+	if (use_chain_dma)
+		return 0;
```

Comment:
How about eliminate the use of use_chain_dma variable and bring
```
if (pipeline->use_chain_dma)
   return 0;
```
to the beginning of the for_each_pcm_streams loop? so the code that make no sense under `pipeline->use_chain_dma == true` is not executed.

Reply:
I thought about that, we can do it, but I wanted to have the smallest change possible which can be squashed into the offending commit.
A bigger change should be a separate one and it should be moved earlier as a simplification for the ChainDMA handling. For upstream, that is.